src/util.ts

```diff
@@ -409,11 +411,15 @@ function encode(value: any): string {
  * @param value
  * @returns {number|null}
  */
-export function hashCode(value: any): number | null {
+export function hashCode(value: any, customHashFunction?: CustomHashFunction): number | string | null {
   if (isNil(value)) return null
 
   let hash = 0
   let s = encode(value)
+	if(customHashFunction) {
+		return customHashFunction(value)
+	}
+
   let i = s.length
```

Comment:
**recommendation:** I suggest moving the default hashing algorithm into a constant say `DEFAULT_HASH_FUNC` and using that as a default argument on the `hashCode` function.

```typescript
const DEFAULT_HASH_FUNC = ...

export function hashCode(value: any, hashFunction: HashFunction = DEFAULT_HASH_FUNC)
```

Reply:
done I've change the HashFunction signature to include the encoding part in the hash function